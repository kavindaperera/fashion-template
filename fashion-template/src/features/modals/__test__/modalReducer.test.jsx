import * as types from '../modalConstants';
import reducer from '../modalReducer';
import { initialState } from '../modalReducer';

describe('modal reducer', () => {
    it('should return the initial state', () => {
        expect(reducer(undefined,{})).toEqual(null)
    })
    it('should handle MODAL_OPEN', () => {
        expect(reducer(initialState,{
            type: types.MODAL_OPEN,
            payload: {
                modalType: 'test',
                modalProps: undefined
            }
        })).toEqual({
            modalType: 'test',
            modalProps: undefined
        })
    })
    it('should handle MODAL_OPEN', () => {
        expect(reducer(initialState,{
            type: types.MODAL_OPEN,
            payload: {
                modalType: 'test',
                modalProps: {data: 'testData'}
            }
        })).toEqual({
            modalType: 'test',
            modalProps: {data: 'testData'}
        })
    })
    it('should handle MODAL_OPEN snapshot 1', () => {
        expect(reducer(initialState,{
            type: types.MODAL_OPEN,
            payload: {
                modalType: 'test',
                modalProps: undefined
            }
        })).toMatchSnapshot()
    })
    it('should handle MODAL_OPEN snapshot 2', () => {
        expect(reducer(initialState,{
            type: types.MODAL_OPEN,
            payload: {
                modalType: 'test',
                modalProps: {data: 'testData'}
            }
        })).toMatchSnapshot()
    })


    it('should handle MODAL_CLOSE', () => {
        expect(reducer(initialState,{
            type: types.MODAL_CLOSE,
        })).toEqual(null)
    })

    it('should handle MODAL_CLOSE snapshot 1', () => {
        expect(reducer(initialState,{
            type: types.MODAL_CLOSE,

        })).toMatchSnapshot()
    })
})