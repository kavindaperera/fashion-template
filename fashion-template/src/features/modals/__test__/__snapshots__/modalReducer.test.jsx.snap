// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal reducer should handle MODAL_CLOSE snapshot 1 1`] = `null`;

exports[`modal reducer should handle MODAL_OPEN snapshot 1 1`] = `
Object {
  "modalProps": undefined,
  "modalType": "test",
}
`;

exports[`modal reducer should handle MODAL_OPEN snapshot 2 1`] = `
Object {
  "modalProps": Object {
    "data": "testData",
  },
  "modalType": "test",
}
`;
